src/sentry/static/sentry/app/components/events/interfaces/debugMeta/debugFileDetails/index.tsx

```diff
@@ -87,18 +87,42 @@ class DebugFileDetails extends AsyncComponent<Props, State> {
       return candidates;
     }
 
-    return candidates.map(candidate => {
-      if (!debugFiles.find(debugFile => debugFile.id === candidate.location)) {
+    // Check for unapplied debug files
+    const candidateLocations = candidates
+      .map(candidate => candidate.location)
+      .filter(candidate => !!candidate);
```

Comment:
This is still an array, so lookups are going to have an `O(n)` time complexity, where lookups in a `Set` should have a complexity closer to `O(1)`